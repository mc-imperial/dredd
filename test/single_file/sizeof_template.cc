template <typename a> void b() { a(sizeof(a)); }
