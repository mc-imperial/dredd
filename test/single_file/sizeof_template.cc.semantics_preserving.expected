#include <cinttypes>
#include <cstddef>
#include <functional>
#include <string>


#ifdef _MSC_VER
#define thread_local __declspec(thread)
#elif __APPLE__
#define thread_local __thread
#endif

#include <limits>
#include <cmath>
static thread_local unsigned long long int no_op = 0;

#define MUTATION_RETURN(arg) actual_result
#define REPLACE_EXPR_NOT(args) if ((args) != actual_result) no_op++
#define REPLACE_EXPR_LNOT(args) if ((args) != actual_result) no_op++
#define REPLACE_EXPR_INT_ZERO(args) if ((args) != actual_result) no_op++
#define REPLACE_EXPR_INT_ONE(args) if ((args) != actual_result) no_op++
#define MUTATION_PRELUDE(arg,type) type actual_result = (arg)
static unsigned long __dredd_replace_expr_unsigned_long(unsigned long arg, int local_mutation_id) {
  MUTATION_PRELUDE(arg,unsigned long);
  REPLACE_EXPR_LNOT(arg);
  REPLACE_EXPR_NOT(arg);
  REPLACE_EXPR_INT_ZERO(0);
  REPLACE_EXPR_INT_ONE(1);
  return MUTATION_RETURN(arg);
}

template <typename a> void b() { a(__dredd_replace_expr_unsigned_long(sizeof(a), 0)); }
